### DISCUSSION TITLE
Configure TRÅDFRI Bulb On/Off behaviour on power cycle

### DISCUSSION THREAD

**Discussion by brentk7** [Koenkk/zigbee2mqtt#6492] [Devices]
According to the [latest Ikea release notes](https://ww8.ikea.com/ikeahomesmart/releasenotes/releasenotes.html), one of the new features in their gateway 1.13.21 is: "Configure TRÅDFRI Bulb On/Off behaviour on power cycle".
This means that the bulbs must somehow support this function. Is there a way to activate this function in the bulbs from zigbee2mqtt?
I have a few bulbs which have woken us up in the middle of the night after a short power failure and would like to set them so they either maintain their last state after power on, or alternatively remain off after a power on and only switch on when instructed to from the zigbee network.

Thanks

**Comment by sjorge** [comment]
I’ve added the power_on_behavior converter earlier this week. To all bulbs not just ikea.

if you are on z2m dev you should je able to configure it. Doesn’t seem to work for all my bulbs though. I think anything on a 2.x firmware should work.

https://github.com/Koenkk/zigbee-herdsman-converters/blob/d67802a8f61bd9ee7aa37c7d0481f03a7db342e3/converters/toZigbee.js#L52

**Reply by brentk7** [reply]
Thanks for that @sjorge I'm running z2m as a  HASS.IO addon. Going to have to figure out if I can update that to the dev branch.

**Reply by brentk7** [reply]
@sjorge I've got the Edge version of z2m running on Hass which I believe should have pulled the dev version - but I don't see anywhere to set power_on_behavior. The light I'm trying is running fw 2.3.050. Are you able to provide any pointers?

**Reply by sjorge** [reply]
If you're on the dev version you should be able to set it using

```mosquitto_pub -t zigbee2mqtt/<friendlyname>/set/power_on_behavior -m 'off'```

And read the current setting using

```mosquitto_pub -t zigbee2mqtt/<friendlyname>/get/power_on_behavior -m ''```

**Reply by brentk7** [reply]
Thanks - that works perfectly.

**Reply by sjorge** [reply]
```
mosquitto_pub -t zigbee2mqtt/<friendlyname>/set/power_on_behavior -m 'on'
mosquitto_pub -t zigbee2mqtt/<friendlyname>/set/level_config -m '{"current_level_startup": "128"}'
```

Some bulbs/panels also take a level_config, so you can also set them to turn on at 50% brightness instead... not all do though, it's more of a hit-and-miss.
